apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/QueryTest.cs

```diff
@@ -334,10 +331,101 @@ public void QueryOnView()
             // This is how a client can check that a BigQueryTable is a view.
             Assert.NotNull(view.Resource.View);
 
-            var queryResults = client.CreateQueryJob($"SELECT * FROM {view} WHERE player = 'Bob'").GetQueryResults().GetRows().ToList();
+            var queryResults = client.ExecuteQuery($"SELECT * FROM {view} WHERE player = 'Bob'").GetRows().ToList();
             Assert.Equal(1, queryResults.Count);
             // The earlier game is not present in the view
             Assert.Equal(85L, (long) queryResults[0]["score"]);
         }
+
+        [Fact]
+        public void BrokenQuery()
+        {
+            BigQueryClient client = BigQueryClient.Create(_fixture.ProjectId);
+
+            string sql = $"This is a broken query";
+            var job = client.CreateQueryJob(sql);
+            Assert.Throws<GoogleApiException>(() => job.GetQueryResults());
+        }
+
+        [Fact]
+        public void NonQueryJob()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            string tableId = _fixture.CreateTableId();
+
+            string[] csvRows =
+            {
+                "Name,GameStarted,Score",
+                "Ben,2014-08-19T12:41:35.220Z,85",
+                "Lucy,2014-08-20T12:41:35.220Z,130",
+                "Rohit,2014-08-21T12:41:35.220Z,90"
+            };
+
+            var bytes = Encoding.UTF8.GetBytes(string.Join("\n", csvRows));
+
+            TableSchema schema = null;
+            var job = client.UploadCsv(_fixture.DatasetId, tableId, schema, new MemoryStream(bytes), new UploadCsvOptions { Autodetect = true });
+
+            Assert.Throws<InvalidOperationException>(() => job.GetQueryResults());
+            Assert.Throws<InvalidOperationException>(() => client.GetQueryResults(job.Reference));
+        }
+
+        [Fact]
+        public void ExecuteQuery_Timeout()
+        {
+            // SQL that I happen to know takes over 10 seconds to query.
+            string sql = "SELECT id FROM [bigquery-public-data:github_repos.contents] where content contains 'NodaTime' AND content contains '2.0.2' LIMIT 1000";
+            BigQueryClient client = BigQueryClient.Create(_fixture.ProjectId);
+            var queryOptions = new QueryOptions { UseLegacySql = true, UseQueryCache = false };
+            var resultsOptions = new GetQueryResultsOptions { Timeout = TimeSpan.FromSeconds(2) };
+            Assert.Throws<TimeoutException>(() => client.ExecuteQuery(sql, queryOptions, resultsOptions));                
+        }
+
+        [Fact]
+        public void GetQueryResults_Timeout()
+        {
+            // SQL that I happen to know takes over 10 seconds to query.
+            string sql = "SELECT id FROM [bigquery-public-data:github_repos.contents] where content contains 'NodaTime' AND content contains '2.0.2' LIMIT 1000";
```

Comment:
Have added a commit to skip these tests by default. We can still run them when we want to.